kiwi/storage/subformat/vagrant_base.py

```diff
@@ -0,0 +1,266 @@
+#
+import json
+import os
+import random
+from textwrap import dedent
+from tempfile import mkdtemp
+from string import Template
+from typing import Dict, Optional  # pylint: disable=unused-import
+
+from kiwi.storage.subformat.base import DiskFormatBase
+from kiwi.command import Command
+from kiwi.system.result import Result  # pylint: disable=unused-import
+
+from kiwi.exceptions import (
+    KiwiFormatSetupError
+)
+
+
+def get_vagrantfile_template(custom_settings=None):
+    # type: (Optional[str]) -> Template
+    """
+    **Generate a Vagrantfile configuration template**
+
+    This function creates a simple template for the Vagrantfile that is
+    included inside a vagrant box.
+
+    The included Vagrantfile carries additional information for vagrant: by
+    default that is just the boxes' MAC address, but depending on the provider
+    additional information need to be present. These can be passed as the
+    parameter ``custom_settings``.
+
+    :param custom_settings: String of additional settings that get pasted into
+        the Vagrantfile template. The string is put at the correct indentation
+        level for you, but the internal indentation has to be provided by the
+        caller.
+    :type custom_settings: str.
+    :returns: A :class:`string.Template` with ``custom_settings`` inserted at
+        the appropriate position. The template has one the variable
+        ``mac_address`` that must be substituted.
+
+    :Example:
+
+    The default without any additional settings will result in this
+    Vagrantfile:
+    >>> print(get_vagrantfile_template().substitute({'mac_address': 'deadbeef'}))
+    Vagrant.configure("2") do |config|
+      config.vm.base_mac = "deadbeef"
+    end
+
+    If your provider/box requires additional settings, provide them as follows:
+    >>> extra_settings = dedent('''
+    ... config.vm.hostname = "no-dead-beef"
+    ... config.vm.provider :special do |special|
+    ...   special.secret_settings = "please_work"
+    ... end
+    ... ''').strip()
+    >>> print(get_vagrantfile_template(extra_settings).substitute({'mac_address': 'DEADBEEF'}))
+    Vagrant.configure("2") do |config|
+      config.vm.base_mac = "DEADBEEF"
+      config.vm.hostname = "no-dead-beef"
+      config.vm.provider :special do |special|
+        special.secret_settings = "please_work"
+      end
+    end
+    """
+
+    INDENT = '  '
+
+    HEADER = dedent('''
+        Vagrant.configure("2") do |config|
+    ''').strip() + os.linesep
+
+    MAC = INDENT + dedent('''
+        config.vm.base_mac = "${mac_address}"
+    ''').strip() + os.linesep
+
+    END = dedent('''
+        end
+    ''').strip()
+
+    template_data = HEADER
+    template_data += MAC
+    if custom_settings:
+        template_data += INDENT
+        template_data += INDENT.join(
+            custom_settings.splitlines(True)
+        )
+        template_data += os.linesep
+    template_data += END
+    return Template(template_data)
```

Comment:
Could we please put the template into its own class. We use this concept in kiwi in other places too. See e.g ```kiwi/storage/subformat/template/vmware_settings.py``` or ```kiwi/bootloader/template/```